File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    let stack = [];
    moves.forEach((item) => {
        for(let i = 0; i < board.length; i++){
            if(board[i][item-1] === 0){
                continue;
            }
            stack.push(board[i][item-1]);
            board[i][item-1] = 0;
            break;
        }
    })
    let result_stack = [...stack]
    for(let i = 0; ; i++){
        if(i >= result_stack.length){
            break;
        }
        if(result_stack[i] === result_stack[i+1]){
            result_stack.splice(i, 2);
            i = -1;
        }
    }
    return stack.length - result_stack.length
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    basket = []
    score = 0

    for i in moves:
        for j in range(len(board)):
            if board[j][i - 1] != 0:
                doll = board[j][i - 1]

                if len(basket) == 0:
                    basket.append(doll)

                elif basket[-1] != doll:
                    basket.append(doll)

                else:
                    basket.pop()
                    score += 2

                board[j][i - 1] = 0
                break

    return score